pkg/queue/manager.go

```diff
@@ -206,6 +206,14 @@ func (m *Manager) Pending(cq *kueue.ClusterQueue) int32 {
 	return int32(len(m.clusterQueues[cq.Name].heap.heap))
 }
 
+func (m *Manager) QueueExists(wl *kueue.QueuedWorkload) (*Queue, bool) {
```

Comment:
Once you move `ClusterQueueExists` here, perhaps you don't need to return the Queue.

Alternatively, a single function could return 2 bools, for whether the queue and clusterqueue exist.

Reply:
The return helps to log the proper non existen queue or clusterQ